packages/date-picker/src/vaadin-date-picker-mixin.js

```diff
@@ -645,27 +656,65 @@ export const DatePickerMixin = (subclass) =>
       this._shouldKeepFocusRing = focused && this._keyboardActive;
     }
 
-    /** @private */
-    __dispatchChange() {
-      this.validate();
-      this.dispatchEvent(new CustomEvent('change', { bubbles: true }));
+    /**
+     * Depending on the type of value change that has taken place
+     * since the last commit attempt, triggers validation and fires
+     * an event:
+     *
+     * ```text
+     * +--------------------------+-------------------+
+     * | Type of value change     | Event             |
+     * +--------------------------+-------------------+
+     * | empty => parsable        | change            |
+     * | parsable => empty        | change            |
+     * | parsable => parsable     | change            |
+     * | parsable => unparsable   | change            |
+     * | empty => unparsable      | unparsable-change |
+     * | unparsable => empty      | unparsable-change |
+     * | unparsable => parsable   | unparsable-change |
+     * | unparsable => unparsable | unparsable-change |
+     * +--------------------------+-------------------+
+     * ```
+     *
+     * If no value change is detected, the method returns false.
+     *
+     * @private
+     * @return {boolean} whether a change was detected and committed.
+     */
+    __commitValueChange() {
+      let result = false;
+
+      if (this.__committedValue !== this.value) {
+        result = true;
+        this.validate();
+        this.dispatchEvent(new CustomEvent('change', { bubbles: true }));
+      } else if (this.__committedUnparsableValue !== this._unparsableValue) {
+        result = true;
+        this.validate();
+        this.dispatchEvent(new CustomEvent('unparsable-change'));
```

Comment:
Yeah, we've considered this option too, but in that case we will need to pass it also with the `change` event. Note, the component fires `change` but not `unparsable-change` when something parsable changes to unparsable and vice versa. So, ultimately, we come to the same outcome as we get with property synchronization in Flow. In this light, property synchronization appears to be a better option because it doesn't require changing the change event's data.

Reply:
Understandable, but wouldn't that be a good change anyway? This would align the change event also to the change event from the normal inputs where you can access the value https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event

Reply:
You are right, dismiss my comment 😬 it's done by accessing the event.target.value instead of details